### PR TITLE
feat: make the clipboard dependency optional

### LINKED ISSUE #1559: Change Clipboard such that it is imported from @react-native-clipboard/clipboard
**Describe the bug**

Clipboard is extracted from the package `react-native` and is introduced as a separate package under `@react-native-clipboard/clipboard`.

You will get the warning often when you use the previous import as:

```
 WARN  Clipboard has been extracted from the react-native core and will be removed in a future release. It can now be installed and imported from '@react-native-clipboard/clipboard' instead of 'react-native'. See https://github.com/react-native-clipboard/clipboard
```

**Expected behavior**
Change the usage of the Clipboard such that it is used from the appropriate package.

### PR DESCRIPTION
**Pull request by madsroskar**
## 🎯 Goal

This PR does a few things: 

1. Make the Clipboard dependency optional on RN CLI
2. Make the Clipboard dependency optional on Expo
3. Separates expo native handlers into separate files so it's structured like RN CLI handlers

This fixes #1559

## 🛠 Implementation details

Instead of using the deprecated clipboard from `react-native`, I've set it up as a native handler, and made it optional to install it for both flavors of the SDK

## 🎨 UI Changes

None interesting enough to screenshot, but if the clipboard dependencies aren't installed, the `copy message` message action doesn't display. Afaik, that's the only place we use it at the moment.

### RN CLI

1. Run an app without the clipboard dependency (see the documentation changes)
2. Check that you don't get the `Copy message` action when you long press a message
3. install the clipboard dependency
4. Check that you **do** get the `Copy message` action when you long press a message

### Expo

1. Run an app without the clipboard dependency (see the documentation changes)
2. Check that you don't get the `Copy message` action when you long press a message
3. install the clipboard dependency
4. Check that you **do** get the `Copy message` action when you long press a message

## ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Documentation is updated
- [x] New code is tested in main example apps, including all possible scenarios
  - [x] SampleApp iOS and Android
  - [x] Expo iOS and Android

### CODE REVIEW

File: package/native-package/src/handlers/setClipboardString.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+let Clipboard: { setString: (string: string) => void };
```
[the remark this answers is not in the document]
Comment (madsroskar) [reply package/native-package/src/handlers/setClipboardString.ts:1]: I'm not sure I follow what you mean - could you elaborate?

File: package/expo-package/src/handlers/index.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+export * from './compressImage';
+export * from './deleteFile';
```
[the remark this answers is not in the document]
Comment (madsroskar) [reply package/expo-package/src/handlers/index.ts:2]: That's an oopsie, file is empty. Probably missing a `:w` lol. Fixing it, nice catch

File: package/expo-package/src/handlers/shareImage.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import * as Sharing from 'expo-sharing';
+
+export const shareImage = async ({ type, url }: { type: string; url: string }) => {
+  try {
+    await Sharing.shareAsync(url, { mimeType: type, UTI: type });
+    return true;
+  } catch (error) {
+    throw new Error('Sharing failed or cancelled...');
```
[the remark this answers is not in the document]
Comment (madsroskar) [reply package/expo-package/src/handlers/shareImage.ts:8]: I'm not sure, I just kept the functionality as it was as much as possible. I can change it to a warning intead

File: package/expo-package/src/handlers/setClipboardString.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+let Clipboard: { setString: (string: string) => void } | undefined;
```
[the remark this answers is not in the document]
Comment (madsroskar) [reply package/expo-package/src/handlers/setClipboardString.ts:1]: I'm not sure - should we? We're already kind of cluttering up the metro logs so I wasn't sure if we wanted to do that.

File: package/expo-package/src/handlers/setClipboardString.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+try {
+  Clipboard = require('expo-clipboard');
+  console.log({ Clipboard });
+} catch (e) {
+  // do nothing
+}
+
+export const setClipboardString =
+  Clipboard !== undefined ? (string: string) => Clipboard.setString(string) : null;
```
Comment (vanGalilea) [inline package/expo-package/src/handlers/setClipboardString.ts:11]: > should we warn and prompt the user to install the opt. dep. ?

```suggestion
  Clipboard !== undefined ? (string: string) => Clipboard.setString(string) : (string: string) => console.warn("Please install expo-clipboard to use the copy message functionaity");
```

File: package/expo-package/src/handlers/setClipboardString.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (vanGalilea) [reply package/expo-package/src/handlers/setClipboardString.ts:1]: Let's keep it clean, a client asked a while ago how to remove the video warning. Let's go for the silent approach.

File: package/native-package/src/handlers/setClipboardString.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (vanGalilea) [reply package/native-package/src/handlers/setClipboardString.ts:1]: Not relevant we're going for the silent approach- not bugging the user if he/she didn't install an optional package

File: package/expo-package/src/handlers/shareImage.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (vanGalilea) [reply package/expo-package/src/handlers/shareImage.ts:8]: Let's indeed take the less aggressive approach here and warn.

[the remark this answers is not in the document]
Comment (santhoshvai) [reply package/expo-package/src/handlers/shareImage.ts:8]: I would say we just warn here. Throw is not useful and wrong in my opinion

[the remark this answers is not in the document]
Comment (madsroskar) [reply package/expo-package/src/handlers/shareImage.ts]: Changed it for both expo and CLI

**Review by vanGalilea — APPROVED** [review]

**Pull request merged**